src/wr_logic_longrange/nodes/finder.py

```diff
@@ -336,6 +218,80 @@ def get_valley(
     return best_valley
 
 
+
+def get_obstacle_list(
+        sector_angle: float,
+        threshold: float,
+        hist: list
+    ) -> list:
+    """
+    Using the laserscan data coming in, it will join objects together to make an obstacle list. This 
+    will then be used to help find the best valley for the rover to go through. Can be varied through
+    changing the threshold (how close the obstacles need to be to get fit into the obstacle list)
+
+    @param sector_angle (float): The number of degrees contained within one sector
+    @param threshold (float): The threshold distance which triggers obstacle avoidance (in meters)
+    @param hist (list): The LiDAR reading data as a list
+    @return List[List[int]]: The list of left bounds and right bounds on each obstacle
+    """
+    # this will be an array of two values with the beginning and ending index of the obstacle.
+    obstacle_list = list()
+
+    one_obstacle = []
+    for i in range(len(hist)):
+        # This prevents single noisy points from blocking out large portions of the drive window
+        if hist[i] < threshold:
+            one_obstacle.append(i)
+        elif len(one_obstacle) > 1:
```

Comment:
After re-reading the logics, I believe here it should be `len(one_obstacle) >= 1`, unless we specifically want to filter out obstacles smaller than a certain width in the case of noisy data, in which case this width should be a parameter or field.
```suggestion
        elif len(one_obstacle) >= 1:
```